### PR TITLE
feat: prelogin page that lists orgs that were logged into, using cookies

### PR DESCRIPTION
**Pull request by kimskimchi**
## Summary

image

### CODE REVIEW

File: ui/pages/login/callback.js [hunk 1]
```diff
@@ -28,8 +33,20 @@ export default function Callback() {
       } else {
         router.replace('/')
       }
-
       window.localStorage.removeItem('next')
+
+      let visitedOrgs = cookies.get('orgs') || []
```
Comment (BruceMacD) [inline ui/pages/login/callback.js:38]: Just a heads up, we make cookies secure by default on the server at the moment when the request is using https. You'll have to add some logic to not do this for orgs cookies for this to work.

File: ui/pages/prelogin/index.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      {organizations?.length > 0 && (
+        <>
+          <Organizations organizations={organizations} />
+        </>
+      )}
+    </>
+  )
+}
+
+function Organizations({ organizations }) {
```
Comment (jmorganca) [inline ui/pages/prelogin/index.js:24]: Better to keep this in the above component since it isn't re-used anywhere.

File: ui/pages/login/index.js [hunk 3]
```diff
@@ -113,6 +115,14 @@ export default function Login() {
 
       await mutate('/api/users/self')
       router.replace('/')
+      let visitedOrgs = cookies.get('orgs') || []
+      if (visitedOrgs.findIndex(x => x.url === window.location.origin) === -1) {
```
Comment (jmorganca) [inline ui/pages/login/index.js:119]: can use:

```
visitedOrgs.find(o => o.url === window.location.origin)
```

File: ui/pages/prelogin/index.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
+import LoginLayout from '../../components/layouts/login'
+import Cookies from 'universal-cookie'
+
+export default function PreLogin() {
+  const cookies = new Cookies()
+  const organizations = cookies.get('orgs')
+  console.log(organizations)
```
Comment (jmorganca) [inline ui/pages/prelogin/index.js:7]: Don't forget to remove this console.log :-)

File: ui/pages/prelogin/index.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+function Organizations({ organizations }) {
+  return (
+    <>
+      {organizations?.map(o => (
+        <a
+          href={`${o.url}`}
+          key={o.name}
+          className='mt-1 mb-1 w-full rounded-lg border border-violet-300 px-4 py-3 text-center text-2xs text-violet-100 hover:border-violet-100 disabled:pointer-events-none disabled:opacity-30'
+        >
+          {o.url}
```
Comment (jmorganca) [inline ui/pages/prelogin/index.js:33]: This should is only the domain (e.g. no `http://` or `https://` as in the screenshot) right?

File: ui/pages/prelogin/index.js (hunk 4 above)
Comment (kimskimchi) [reply ui/pages/prelogin/index.js:7]: done

File: ui/pages/prelogin/index.js (hunk 2 above)
Comment (kimskimchi) [reply ui/pages/prelogin/index.js:24]: done, see https://github.com/infrahq/infra/pull/3142/checks

File: ui/pages/prelogin/index.js (hunk 5 above)
Comment (kimskimchi) [reply ui/pages/prelogin/index.js:33]: resolved in chat

**Comment by kimskimchi** [comment]
moved to https://github.com/infrahq/infra/pull/3142

File: ui/pages/login/callback.js (hunk 1 above)
Comment (kimskimchi) [reply ui/pages/login/callback.js:38]: resolved in chat

**Pull request closed without merging**